pkgs/development/tools/misc/libtool/libtool2.nix

```diff
@@ -34,10 +34,18 @@ stdenv.mkDerivation rec {
     automake
     autoconf
     popd
+  '' +
+  # libtool commit da2e352735722917bf0786284411262195a6a3f6 changed
+  # the shebang from `/bin/sh` (which is a special sandbox exception)
+  # to `/usr/bin/env sh`, meaning that we now need to patch shebangs
+  # in libtoolize and ltmain.sh since `dontPatchShebangs` is set:
+  ''
+    substituteInPlace libtoolize.in       --replace '#! /usr/bin/env sh' '#!${runtimeShell}'
+    substituteInPlace build-aux/ltmain.in --replace '#! /usr/bin/env sh' '#!${runtimeShell}'
```

Comment:
`substituteInPlace build-aux/ltmain.in --replace '#! /usr/bin/env sh' '#!${runtimeShell}'` now leaks out `/nix/store` paths into tarballs generated under nixpkgs. `build-aux/ltmain.sh` gets packaged as part of `make dist` in `libtool-aware` projects. Encountered it today on `ski`. Executable reproducer:

```
$ git clone https://github.com/trofi/ski && cd ski
$ git checkout v1.4.0
$ nix-shell --pure -p autoconf -p lzip -p automake -p libtool -p pkg-config -p libelf -p ncurses -p gperf -p bison -p flex
$$ ./autogen.sh && ./configure && make dist
$$ tar xf ski-1.4.0.tar.xz
$$ fgrep -R  nix/store ski-1.4.0/
ski-1.4.0/build-aux/ltmain.sh:#!/nix/store/3j918i1nbwhby0y38bn2r438rjhh8f4d-bash-5.1-p16/bin/bash
```

Note how `ski-1.4.0/build-aux/ltmain.sh` embeds the path to `/nix/store`.

Reply:
Thank you for catching this.

I'm investigating a few solutions.

Reply:
I think the fix is

https://github.com/amjoseph-nixpkgs/nixpkgs/commit/b5aad8addf1d4a84dbf7545965584bd19fb7e42e

Testing now...